cico_run_tests.sh

```diff
@@ -32,16 +32,20 @@ make docker-start
 make docker-deps
 make docker-generate
 make docker-build
-make docker-test-unit
 
+# Run the test without generating coverage information (for faster feedback
+# cycles).
+make docker-test-unit-no-coverage
 make integration-test-env-prepare
-
 function cleanup {
   make integration-test-env-tear-down
 }
 trap cleanup EXIT
-
 make docker-test-migration
+make docker-test-integration-no-coverage
+
+# Run the tests that generate coverage information
+make docker-test-unit
```

Comment:
If we split this out, we essentially could double the amount of time it takes for our CI to build and generate the coverage. Maybe we can re-use the docker image from the test job (the one without coverage) in the job that just generates the coverage. @kbsingh would that be possible?